lib/run-compilation.js

```diff
@@ -14,6 +14,7 @@ var errorMessage = require('./constants').errorMessage;
 module.exports = function runCompilation (options, compiler, done) {
   var errors = [];
   var warnings = [];
+  var compilerMethod = options.emitErrors ? 'after-compile' : 'after-emit';
```

Comment:
One minor update - I'm hitting some issues developing this internally at work since we rely on artifactory/Jfrog (e.g. an enterprise/internally managed clone npm). Consequently I'm hitting issues with escaping our proxy to temporarily point @ npm's public registry when we dynamically require/download versions of webpack. I got it to work nonetheless by just manually downloading the packages and porting them to the appropriate locations. This doesn't make testing locally though so I'll have to devote some time to these test cases outside of work hours. I'll tinker around with it hopefully over the next week or so at home when I have time.